### ISSUE TITLE
Linux Cmake error

### ISSUE THREAD

**Issue by Guozht** [weidai11/cryptopp#495]
```
[root@localhost Cryptopp]# make
[  0%] Building CXX object CMakeFiles/cryptopp-object.dir/cryptlib.cpp.o
/tmp/ccS7ThcX.s: Assembler messages:
/tmp/ccS7ThcX.s:1248: Error: no such instruction: `shrx %ecx,%ebp,%edx'
/tmp/ccS7ThcX.s:1270: Error: no such instruction: `shlx %ebx,%r13d,%r13d'
make[2]: *** [CMakeFiles/cryptopp-object.dir/cryptlib.cpp.o] Error 1
make[1]: *** [CMakeFiles/cryptopp-object.dir/all] Error 2
make: *** [all] Error 2
[root@localhost Cryptopp]# 
```

**Comment by noloader** [comment]
Thanks @Guozht.

This looks like an assembler problem:

```
/tmp/ccS7ThcX.s:1248: Error: no such instruction: `shrx %ecx,%ebp,%edx'
/tmp/ccS7ThcX.s:1270: Error: no such instruction: `shlx %ebx,%r13d,%r13d'
```

Its like you have (1) new hardware, (2) new compiler that can generate new intructions; but then you have (3) old assemblers and (4) old linkers.

Are you on a VM? If so, what is the OS and compiler versions? Also, what is the version of Binutils?

Recently we removed `-march=native` as default in the build systems. You may have better luck working from Master (unless you can update Binutils).

**Comment by Guozht** [comment]
problem has been fixed,gcc-version 6.1.0 ,I removed -march=native in makefile. so it looks like a gcc problem.

**Comment by noloader** [comment]
Last call for CMake patches before removing it from the library and adding it to the patch page. If you have something to add, then now is the time to do it.

Also see [Remove CMake from GitHub; add it to Patch Page on wiki?](https://groups.google.com/forum/#!topic/cryptopp-users/PKlz8-ClXOU) and [Commit 1c740b0a097aecaa](https://github.com/weidai11/cryptopp/commit/1c740b0a097aecaa66044036c0d2c54220eda5d5).
